src/modules/session/thunks.ts

```diff
@@ -0,0 +1,41 @@
+import { createAsyncThunk } from "@reduxjs/toolkit";
+import { extendedApi as sessionApi } from "./api";
+import { UpdateProfileRequest } from "./types";
+
+const { origin } = window.location;
```

Comment:
Had to create some thunks (basically just async / await functions that can be dispatched like actions) in order to implement some sequential API call functionality.

I think that as long as we only use thunks to sequentially make API calls, and not to dispatch any actions that would directly update a reducer, they should work well with the RTK Query implementation.

For example, instead of calling

```
const response = await thunkAPI.dispatch(exampleApiCall())
thunk.dispatch(actions.receiveData(response.data);

// do something else
```
we should just call

```
const response = await thunkAPI.dispatch(exampleApiCall());

// do something else
```